modules/compression/include/ips_jpg_at_model.cpp

```diff
@@ -0,0 +1,226 @@
+#include <systemc.h>
+
+/**
+ * @brief jpg_output module. Federico Cruz
+ * It takes the image and compresses it into jpeg format
+ * It is done in 4 parts:
+ * 1. Divides the image in 8x8 pixel blocks; for 8-bit grayscale images the a level shift is done by substracting 128 from each pixel.
+ * 2. Discrete Cosine Transform (DCT) of the 8x8 image.
+ * 3. Each transformed 8x8 block is divided by a quantization value for each block entry.
+ * 4. Each quantized 8x8 block is reordered by a Zig-Zag sequence into a array of size 64.
+ * *5. Entropy compression by variable length encoding (huffman). Used to maximize compression. Not implemented here.
+ */
+#define PI 3.1415926535897932384626433832795
+#define Block_rows 8
+#define Block_cols 8
+
+SC_MODULE (jpg_output) {
+  
+  //input signals
+  sc_in<sc_int<32> > PixelValue_signal;
```

Comment:
Use consistent notation for methods and variable names like camelCase, snake_case, or PascalCase.